src/context/AuthContext.jsx

```diff
@@ -0,0 +1,148 @@
+import axios from "axios";
+import React, { useEffect, useState, createContext, useContext } from "react";
+import { useNavigate } from "react-router-dom";
+import { useToast } from "./ToastContext";
+
+const AuthContext = createContext();
+const initialState = {
+  loading: true,
+  data: [],
+  error: "",
+};
+const AuthProvider = ({ children }) => {
+  const { setToast } = useToast();
+  const navigate = useNavigate();
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [user, setUser] = useState(initialState);
+  const [loginCred, setLoginCred] = useState({ email: "", password: "" });
+  const [signupCred, setSignupCred] = useState({
+    email: "",
+    password: "",
+    confirmPassword: "",
+    firstName: "",
+    lastName: "",
+  });
+
+  const signUp = (e) => {
+    e.preventDefault();
+    if (signupCred.password !== signupCred.confirmPassword) {
+      setToast({
+        show: true,
+        content: "Passwords do not match",
+        type: "info",
+      });
+      return;
+    }
+    axios
+      .post(
+        "/api/auth/signup",
+        JSON.stringify({
+          firstName: signupCred.firstName,
+          lastName: signupCred.lastName,
+          email: signupCred.email,
+          password: signupCred.password,
+        })
+      )
+      .then((res) => {
+        setToast({
+          show: true,
+          content: `Welcome, ${res.data.createdUser.firstName}`,
+          type: "info",
+        });
+        setUser({ loading: false, data: res.createdUser, error: "" });
+        setIsLoggedIn(true);
+      })
+      .catch((err) => {
+        if (err.message)
+          setToast({
+            show: true,
+            content: "Email is already registered",
+            type: "error",
+          });
+
+        setUser({ loading: false, data: [], error: err.message });
+      });
+  };
+  const logIn = (e) => {
+    e.preventDefault();
+    axios
+      .post(
+        "/api/auth/login",
+        JSON.stringify({
+          email: loginCred.email,
+          password: loginCred.password,
+        })
+      )
+      .then((res) => {
+        setToast({
+          show: true,
+          content: `Welcome, ${res.data.foundUser.firstName}`,
+          type: "info",
+        });
+        localStorage.setItem("token", res.data.encodedToken);
+        setUser({ loading: false, data: res.data.foundUser, error: "" });
+        setIsLoggedIn(true);
+      })
+      .catch((err) => {
+        if (err.message.slice(err.message.lastIndexOf(" ") + 1) === "401")
+          setToast({
+            show: true,
+            content: "Wrong Password",
+            type: "error",
+          });
+        else if (err.message.slice(err.message.lastIndexOf(" ") + 1) === "404")
+          setToast({
+            show: true,
+            content: "Email is not registered yet",
```

Comment:
suggestion: all error messages could come from an object instead of hard coding it, it will be easier to make changes later

Reply:
As the error message is coming from the backend so I don't have control over the error message.
That's I only checked the last word of the error message

Reply:
i meant error content, example in setToast there is "Email is not registered"